### ISSUE TITLE
First name and Last name combined should be considered during contact duplicate scan

### ISSUE THREAD

**Issue by Iheanacho2027** [SORMAS-Foundation/SORMAS-Project#2405]
### Problem Description
Currently, the system only checks with one variable name and brings out about 40 - 100 names that have no relation to the contact which is about to be imported i.e it doesnt make sense if the system is using age and sex to check for similarity when both first name and last name combined isnt the same, it slows down sessions like import
### Proposed Change
The system should consider both first and last name combined when scanning for duplicate person

**Comment by bernardsilenou** [comment]
@MateStrysewskeSym  I think this point was solved in an earlier issue, or not?
Do we need more refinement on the method for duplicate person detection?
Maybe we need a dynamic algorithm that only shows the top 5 most similar matches

**Comment by MateStrysewskeSym** [comment]
That's right, there's already an issue for this and it will be fixed in the next update.

**Comment by MateStrysewskeSym** [comment]
@bernardsilenou I think what @Iheanacho2027 means is not that the order of the names should not matter, but that the duplicate detection should be more restrictive. We made it pretty forgiving on purpose to account for spelling mistakes or missing second names, but since it apparently results in too many false duplicates, we should probably change this logic.

**Comment by ChristopherRiedelSymeda** [comment]
Maybe we should consider to make the name similarity threshold value customizable in sormas.properties

**Comment by bernardsilenou** [comment]
> @bernardsilenou I think what @Iheanacho2027 means is not that the order of the names should not matter, but that the duplicate detection should be more restrictive. We made it pretty forgiving on purpose to account for spelling mistakes or missing second names, but since it apparently results in too many false duplicates, we should probably change this logic.

Thanks, understood

**Comment by bernardsilenou** [comment]
> Maybe we should consider to make the name similarity threshold value customizable in sormas.properties

This is good idea. 
@MateStrysewskeSym since first and last name are required for all entities, then I agree with you that should consider both names in determining the similarity measure. The tolerance should only account for spelling errors. Thus similarity measure will be independent of name order and capital letters. After theses changes we can then increase the threshold value to 0.95, because in reality, spelling error for names occur only 5% of the time at the population level.

**Comment by bernardsilenou** [comment]
> Maybe we should consider to make the name similarity threshold value customizable in sormas.properties

@ChristopherRiedelSymeda  What do you mean by 'customizable' please?  Is this different from what we already have that teh user can set the value 0 to 1
